components/automate-cli/pkg/verifyserver/services/softwareversionservice/softwareversionservice.go

```diff
@@ -0,0 +1,203 @@
+package softwareversionservice
+
+import (
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/fiberutils"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/getosutils"
+	"github.com/chef/automate/lib/logger"
+	"github.com/pkg/errors"
+)
+
+type ISoftwareVersionService interface {
+	GetSoftwareVersionServices(string) (*models.SoftwareVersionDetails, error)
+}
+
+type SoftwareVersionService struct {
+	cmdCheckArray []string
+	osFilepath    string
+	logger        logger.Logger
+	CheckPath     func(cmd string) (string, error)
+}
+
+func NewSoftwareVersionService(logger logger.Logger, CheckPath func(cmd string) (string, error)) ISoftwareVersionService {
+	return &SoftwareVersionService{
+		cmdCheckArray: cmdCheckArray,
+		osFilepath:    OSFILEPATH,
+		logger:        logger,
+	}
+}
+
+const (
+	AVAILABILITY              = " availability"
+	ENSURE                    = "Ensure "
+	OSFILEPATH                = "/etc/os-release"
+	LINUX_VERSION_CHECK       = "Linux Version Check"
+	UBUNTU                    = "Ubuntu"
+	RED_HAT                   = "Red Hat"
+	OPENSEARCH                = "opensearch"
+	AUTOMATE                  = "automate"
+	POSTGRES                  = "postgres"
+	CHEF_SERVER               = "chef-server"
+	BASTION                   = "bastion"
+	DEBIAN                    = "Debian"
+	RED_HAT_SUPPORTED_VERSION = 7
+)
+
+var cmdCheckArray = []string{"mkdir", "useradd", "chown", "rm", "touch", "truncate", "echo", "sleep", "ls", "grep", "yum", "which", "cp", "curl", "bash", "sysctl", "cat", "sed", "mount", "mv", "systemd", "wget", "exec", "rsync"}
+
+func (sv *SoftwareVersionService) GetSoftwareVersionServices(query string) (*models.SoftwareVersionDetails, error) {
```

Comment:
I think you can rename this function as GetSoftwareVersionDetails as there is no services that you are returning so having services in func name makes no sense